azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -201,15 +215,44 @@ private boolean isMultiExecutorMode() {
    */
   private void refreshExecutors() {
     synchronized (activeExecutors) {
-      // TODO: rest api call to refresh executor stats
+      ExecutorService taskExecutors =
```

Comment:
+1 on reusing.
I was also thinking whether a threadpool's necessary, or just an arraylist would do?  Because you are not using the threadpool to limit parallel execution or anything?